CIP-ledger-evolution/README.md

```diff
@@ -0,0 +1,302 @@
+
+Many thanks Michael Peyton Jones to writing [CIP-35] and forging this path for us.
+Some of the language here has been taken directly from it.
+
+## Background
+
+### Terminology
+
+Context for the terminology used in this document is given in [CIP-59].
+
+### Specifications
+
+The ledger is specified as a state transition system using a
+[small step operational semantics](STS.md).
+We refer to this framework as the *Small Step Semantics for Cardano*, or the *STS* for short.
+An understanding of the existing STS specifications for the
+[existing ledger eras](https://github.com/input-output-hk/cardano-ledger#cardano-ledger)
+is often required to fully understand the implications of changes to the ledger
+(though an understanding of the Haskell implementation is a fair substitute).
+
+The STS framework leaves both cryptographic primitives and the serialization format abstract.
+These specifications need to be complete enough to realize a full implementation of the ledger
+given the cryptographic primitives and serialization format.
+The cryptographic primitives are described as appendices to the STS specifications,
+and the serialization format is given as a
+[CDDL file](https://www.rfc-editor.org/rfc/rfc8610).
+There SHOULD be one STS specification per ledger era
+(the Allegra ledger era is currently combined with the Mary era).
+
+From the Byron to the Babbage ledger eras, the STS frameworks were written in $\LaTeX$.
+Starting in the Conway ledger era, literate Agda will be used.
+During the transition from $\LaTeX$ to literate Agda, we will take advantage
+of the ability to substitute $\LaTeX$ in place of Agda code when needed for expedience.
+With time, the Agda specification will gloriously blur the line between specification
+and reference implementation,
+just as the type system upon which Agda is built blurs the line between proof and program.
+
+### Ledger eras
+
+A ledger era is a collection of features added to the ledger which are introduced at a hard fork.
+The existing ledger eras, with a very simplistic description is given below.
+
+|name|new features|
+| --- | --- |
+|Byron|initial UTxO ledger|
+|Shelley|decentralized block production, stake delegation|
+|Allegra|timelock scripts|
+|Mary|multi-assets|
+|Alonzo|Plutus scripts|
+|Babbage|improved Plutus script contexts|
+|Conway|constitutional committee replacement, SPO voting on hard forks|
+
+
+### Ledger state
+
+The ledger state is the source of truth from which the network derives meaning.
+For performance reasons, we require it to be the minimal state needed
+to derive this value and validate blocks.
+
+There is currently one exception to the minimality, which will be deprecated in the days ahead,
+namely the stake pool ranking information which is used by Daedalus.
+
+#### Ledger events
+
+Some provenance about the ledger calculations is provided by ledger events.
+These events come with zero cost to a running node if not used and are not stored in the ledger state.
```

Comment:
Right, I agree. I just thought it was a useful definition, and later I say that they do not need to be a part of the CIP processes. but later in the PR discussion here, I've argued with myself :) (see below)

Reply:
Yeah, I can see that maybe this is because it's hard for you to pin down _positively_ the things that require CIPs. For Plutus we said "these things require CIPs, most other things shouldn't", whereas for the ledger maybe you feel like you want to say "these things _don't_ require CIPs, everything else does".